lib/protocol/message.go

```diff
@@ -148,6 +156,13 @@ type Device struct {
 	Options         []Option // max:64
 }
 
+type FileDownloadProgressUpdate struct {
```

Comment:
The fact that we are pulling the file implies that we have the have the index entry for it, so I guess we are just trying to make the messages as slim as possible.

I'd argue that the current implementation is not doing the right thing, and it should ideally refer to blocks by indexes/hashes.